src/linker/Linker/MessageContainer.cs

```diff
@@ -256,10 +256,10 @@ public string ToMSBuildString ()
 			if (Origin?.Provider != null) {
 				if (Origin?.Provider is MethodDefinition method)
 					sb.Append (method.GetDisplayName ());
-				else if (Origin?.Provider is TypeDefinition type)
+				else if (Origin?.Provider is MemberReference type)
```

Comment:
nit: probably change the name from type a memberReference could be several things